### PR TITLE
Added Compile validation CI setup to starter project

### PR DESCRIPTION
**Pull request by danielimprobable**
This PR add the updated CI scripts for the  [Starter Project](https://github.com/spatialos/UnrealGDKStarterProject).

Primary reviewers:
@improbable-valentyn @m-samiec

### CODE REVIEW

File: ci/prepare_build_environment.sh [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
-source ci/profiling.sh
 
 if ! isTeamCity ; then
   echo "This script should only be run on the CI agents."
   exit 0
 fi
 
 if ! isWindows ; then
-  echo "StarterProject can only be built on Windows."
+  echo "TestSuite can only be built on Windows."
```
Comment (mironec) [inline ci/prepare_build_environment.sh]: Shouldn't this stay StarterProject?

File: ci/build_linux.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+source ci/profiling.sh
+
+markStartOfBlock "$0"
+
+source ci/prepare_build_environment.sh
+
+markStartOfBlock "Build the StarterProject (Linux)"
+  # Build each target to ensure scripts are correct, skipping code generation on all but the first to save some time.
+  Game/Scripts/BuildWorker.bat "StarterProjectEditor" "Win64" "Development" "StarterProject.uproject"
```
Comment (improbable-valentyn) [inline ci/build_linux.sh]: Why do we need to build Win64 editor both here and in `build_windows.sh`?

Comment (danielimprobable) [reply ci/build_linux.sh]: You need to have the editor built to be able to cross compile for linux

Comment (danielimprobable) [reply ci/build_linux.sh]: At least that was the case in the past, not sure if we have changed anything that would have caused it to change, but I think this is still the case.

**Pull request merged**